### ISSUE TITLE
Unable to remove actions declared by WC_Admin_Post_Types

### ISSUE THREAD

**Issue by scottybo** [woothemes/woocommerce#11918]
## EXPLANATION OF THE ISSUE

Version 2.4.0 of Woocommerce, Version 4.6.1 of Wordpress

The class WC_Admin_Post_Types defines a number of actions and filters in the __construct function. Due to various reasons, I don't want certain logged-in users to be able use the advanced filter on the Orders page (i.e. search for a customer).

The action I'm trying to remove is on line 70 of class-wc-admin-post-types.php

`add_action( 'restrict_manage_posts', array( $this, 'restrict_manage_posts' ) );`

As the class is called without defining a variable, it's (seemingly) impossible for me to remove this action.

I suggest making it possible to manage these hooks by simply changing the "new WC_Admin_Post_Types();" to something along the lines of

```
global $wc_admin_post_types;
$wc_admin_post_types = new WC_Admin_Post_Types();

```
Which would allow me to do something along the lines of
```
add_action('wp_loaded', function() {
    global $wc_admin_post_types;
    remove_action( 'restrict_manage_posts', array($wc_admin_post_types, 'restrict_manage_posts') );
});
```

## SYSTEM STATUS REPORT

n/a

**Comment by mikejolley** [comment]
This seems like an odd thing to remove to me. I get that this cannot be unhooked right now, but it's core UI and really shouldn't be touched by plugins if possible. If it literally just needs to hide the search from the user, CSS can be used instead and that can be enqueued based on user level. Then our admin code can be left as-is.

`.wc-customer-search` css selector.

**Comment by scottybo** [comment]
I appreciate that it might seem odd, but the point still stands :) 

There are a million and one reasons why someone might want to adjust hooks, and it would be great if it were made possible

**Comment by karskiy** [comment]
https://gist.github.com/karskiy/e6d0f456acf41fbf16143c950fd778b1

**Comment by karskiy** [comment]
https://gist.github.com/karskiy/e6d0f456acf41fbf16143c950fd778b1

**Comment by markkap** [comment]
only thing that is possible to say is grrrrrrrrrrrrrrrrrrrr. Being able to remove hooks is the ABC of wordpress. please tell me it was fixed in 3.0.

(I don't have a use for filtering by product categories if it makes a difference)
